File: Data Structures & Algorithms/max-water-container/submission-2.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int maxArea = 0;
        int l = 0;
        int r = heights.Length-1;
        while(l<r){
            int area = (r-l)*Math.Min(heights[l], heights[r]);
            maxArea = Math.Max(maxArea, area);
            if(heights[l]<heights[r]) l++;
            else r--;
        }
        return maxArea;
    }
}

File: Data Structures & Algorithms/maximum-product-subarray/submission-2.java
class Solution {
    public int maxProduct(int[] nums) {
        int curMin = 1, curMax = 1;
        int res = Integer.MIN_VALUE;
        for(int num: nums) res = Math.max(res, num);

        for(int n: nums){
            int tmp = curMax;
            curMax = Math.max(Math.max(n*curMax,n*curMin),n);
            curMin = Math.min(Math.min(n*tmp,n*curMin),n);
            res = Math.max(res, curMax);
        }
        return res;
    }
}
